applications/ananke/src/ananke_listener.erl

```diff
@@ -0,0 +1,294 @@
+normalize_schedule(Schedule) ->
+    Action = wh_json:get_value(<<"action">>, Schedule),
+    ActionType = wh_json:get_value(<<"type">>, Action),
+    ActionFun = action_fun(ActionType, Action),
+    TimeSchedule = time_schedule(Schedule),
+    ActionName = action_name(ActionType, Action, TimeSchedule),
+    {ActionName, TimeSchedule, ActionFun}.
+
+-spec time_schedule(wh_json:object()) -> time().
+time_schedule(Schedule) ->
+    GetTimeTokenFun = get_time_token_value(Schedule),
+    case wh_json:get_value(<<"type">>, Schedule) of
+        <<"every">> ->
+            [Minutes, Hours, MonthDays, Monthes, Weekdays] = lists:map(GetTimeTokenFun
+                                                                       ,[{<<"minutes">>, 'all'}
+                                                                         ,{<<"hours">>, 'all'}
+                                                                         ,{<<"month_days">>, 'all'}
+                                                                         ,{<<"monthes">>, 'all'}
+                                                                         ,{<<"weekdays">>, 'all'}
+                                                                        ]),
+            {'cron', {Minutes, Hours, MonthDays, Monthes, Weekdays}};
+        <<"once">> ->
+            [Second, Minute, Hour, Day, Month, Year] = lists:map(GetTimeTokenFun
+                                                                 ,[{<<"second">>, 0}
+                                                                   ,{<<"minute">>, 0}
+                                                                   ,{<<"hour">>, 0}
+                                                                   ,{<<"day">>, 1}
+                                                                   ,{<<"month">>, 1}
+                                                                   ,{<<"year">>, 1970}
+                                                                  ]),
+            {'oneshot', {{Year, Month, Day}, {Hour, Minute, Second}}};
+        <<"periodic">> ->
+            [Seconds, Minutes, Hours, Days] = lists:map(GetTimeTokenFun
+                                                        ,[{<<"seconds">>, 0}
+                                                          ,{<<"minutes">>, 0}
+                                                          ,{<<"hours">>, 0}
+                                                          ,{<<"days">>, 0}
+                                                         ]),
+            {'sleeper', ?MILLISECONDS_IN_SECOND * (Seconds
+                                                   + ?SECONDS_IN_MINUTE * Minutes
+                                                   + ?SECONDS_IN_HOUR * Hours
+                                                   + ?SECONDS_IN_DAY * Days)}
+    end.
+
+-spec schedule({ne_binary(), time(), fun()}) -> {'ok', pid()} | {'error', term()}.
```

Comment:
ananke_listener.erl:229: Invalid type specification for function ananke_listener:schedule/1. The success typing is ({atom() | binary(),{'cron',{_,_,_,_,_}} | {'oneshot',pos_integer() | {_,_}} | {'sleeper',pos_integer()},{fun(),[any()]} | {atom(),atom(),[any()]}}) -> {'error',_} | {'ok',pid()}